### DISCUSSION TITLE
Update Self Host Budibase from 2.8.10 to 2.19 best procedure question

### DISCUSSION THREAD

**Discussion by gmentzik** [Budibase/budibase#13024] [Help] [answered]
Hi devs,

We would like to update self host Budibase from 2.8.10 to  2.19. 
We have registered self-host Budibase with docker-compose (not single image). 
We noticed at latest docker-compose that Couch DB image has been changed from IBM to Budibase, and curl has been removed.

Please check our questions bellow on update procedure and cli to help us move forward:

Questions: 

1) Shall we use the new docker-compose file? 

2) What changes with Budibase/CouchDB compared to IBM? 

3) Will this new CouchDB need to be updated per release as the 3 core Budibase images from now on or considered independent?

4) If we use the budibase cli it allowing docker-compose update then at update if fetches latest 3 core images plus the new CouchDB image at docker-compose up. 
However at update documentation it says if we used docker compose to create it then to use docker-compose pull which fetches latest for all images in compose files. 
Testing both scenarios Budibase seems to be working fine in both cases after update. 
So is it ok to update to latest docker compose file and pull all latest images including latest MINIO? 
Our current MINIO image is minio/minio 2023-03-24T23:06:39.765123446Z.

Thank you in advance for your prompt response!

**Comment by mike12345567** [comment, marked as answer]
Hi @gmentzik - on your questions:
1. Yes you should try to use the new docker-compose files when updating to the latest versions of Budibase.
2. The change to our image was some efficiency boosts based on Budibase's exact needs of CouchDB, as well as native ARM support. In future we may add some more features to enable better internal table searching capabilities as well.
3. No you shouldn't need to keep updating the Budibase CouchDB image, if there is a change to the image and you haven't updated, the builder will warn you of the incompatibility. 
4. Using the Budibase CLI to update your docker-compose setup is the easiest way to go about this, it is aware of MinIO incompatibility issues and will only update the parts of the docker-compose script which require updates, it doesn't simply replace it. It will also make sure you have the correct CouchDB image being used.

**Comment by gmentzik** [comment]
Ok I understand it is suggested to update docker compose and change go to budibase/couchdb. However i doubt that your no.3 comment as you have introduced pull always at docker compose and this mean every docker compose up will fetch latest.

Regarding MinIO I checked their release notes and there is a number of Bug fixes since image we have, so I understand to not update it every week but maybe every few months?
No. 4 comment is ambigious. docker-compose tags latest for minio as all images. So any docker pull will update to latest as this is a safe action which is aligned to you update instructions page at Budibase documentation.

I have run every upgrade scenario possible :
1. Update through budibase
2. Update with cli with docker-compose update.
3. Update docker compose and docker-compose pull
4. Cli backup -> Install fresh and cli restore.

All work, but want the correct approach.


> Hi @gmentzik - on your questions:
> 
> 1. Yes you should try to use the new docker-compose files when updating to the latest versions of Budibase.
> 2. The change to our image was some efficiency boosts based on Budibase's exact needs of CouchDB, as well as native ARM support. In future we may add some more features to enable better internal table searching capabilities as well.
> 3. No you shouldn't need to keep updating the Budibase CouchDB image, if there is a change to the image and you haven't updated, the builder will warn you of the incompatibility.
> 4. Using the Budibase CLI to update your docker-compose setup is the easiest way to go about this, it is aware of MinIO incompatibility issues and will only update the parts of the docker-compose script which require updates, it doesn't simply replace it. It will also make sure you have the correct CouchDB image being used.



> Hi @gmentzik - on your questions:
> 
> 1. Yes you should try to use the new docker-compose files when updating to the latest versions of Budibase.
> 2. The change to our image was some efficiency boosts based on Budibase's exact needs of CouchDB, as well as native ARM support. In future we may add some more features to enable better internal table searching capabilities as well.
> 3. No you shouldn't need to keep updating the Budibase CouchDB image, if there is a change to the image and you haven't updated, the builder will warn you of the incompatibility.
> 4. Using the Budibase CLI to update your docker-compose setup is the easiest way to go about this, it is aware of MinIO incompatibility issues and will only update the parts of the docker-compose script which require updates, it doesn't simply replace it. It will also make sure you have the correct CouchDB image being used.

**Reply by mike12345567** [reply]
The method we recommend for upgrading is with the Budibase CLI.
